classifier_and_summarizer/summarizer/extractor_output_preprocessor.py

```diff
@@ -0,0 +1,228 @@
+'''This module preprocesses the
+output of the extractor to ready it
+for the later stages of the summarizer
+'''
+
+from nltk.tokenize import sent_tokenize
+from sentence_transformers import SentenceTransformer
+
+from classifier_and_summarizer.summarizer.text_summarizer import TextSummarizer
+from classifier_and_summarizer.summarizer.web_entity_detection import \
+    ImageDescriptionRetriever
+from data_models.contents import ContentType
+
+
+class SentenceWithAttributes:
+    ''' Object to represent a summary sentence
+    or title sentence along with attributes such as
+    embedding or font_style
+    '''
+
+    def __init__(self, text, index, font_style, embedding):
+        self.text = text
+        self.index = index
+        self.font_style = font_style
+        self.embedding = embedding
+
+
+class ExtractorOutputPreprocessor:
+    ''' Class to implement the utilities for
+    preprocessing the extractor output
+
+    Preprocessing steps applied:
+    * Split contents into different types
+    * Assign img_description_embeddings to media
+    * Summarize the text content
+    '''
+
+    def __init__(self, contents):
+        self.content_list = contents.content_list
+        self.normal_text_content_list = list()  # non-title text
+        self.title_text_content_list = list()  # title text
+        self.media_content_list = list()  # images/gifs
+        self.embedded_content_list = list()  # insta/tweets/quotes
+        self.text_summarizer = TextSummarizer(priority="speed")
+        self.sentence_embedding_model \
+            = SentenceTransformer('bert-base-nli-stsb-mean-tokens')
+
+    def get_preprocessed_content_lists(self):
+
+        # Assign 'index' field to every content
+        for content_index in range(len(self.content_list)):
+            self.content_list[content_index].content_index = content_index
+
+        #  first split content into different categories
+        self._split_content()
```

Comment:
I guess, this could be done in `extraction/` module.

Reply:
Thanks for pointing this out! 
Moved this to data_models/contents.py and assigned the field using a counter!